### PR TITLE
Task/compiler updates

### PR DESCRIPTION
**Pull request by rhornung67**
I added new specs based on what is now available on the various testing platforms and corrected a few things where specified versions do not exist.

Questions I have:

- Trying to test some of these new specs manually in RAJA through uberenv, it appears that the new spack we are using (0.19.0)  no longer likes it when you set compiler flags on the command line spec. Is there a new quote-related syntax to use? These flags are set in the compilers.yml file. It would be nice to be able to test different versions of gcc headers with the same compiler, for example.
- The oneapi stuff works when I use it with a host-config file that I manually created. However, through uberenv/spack, it runs into an issue building camp. It looks like the proper gcc headers are not being used in the camp build, but not 100% sure.
- For toss4 (corona) specs, we label the specs based on clang version, not rocm version. Multiple rocm versions report the same clang version for amdclang++. Is there a way to change the specs to be named for rocm version, so we can test if needed?
- SImilarly, for blueos, multipl xl compiler drops, labeled by date, report the same xl compiler version. Is there a way to distinguish by drop date instead of cl version number. For example, 2022.06.25 and 2022.08.19 both report XL Version: 16.01.0001.0012. How can we distinguish?

### CODE REVIEW

**Comment by rhornung67** [comment]
@kab163 please take a look at this PR I put up yesterday. I think this covers the changes in  your PR plus many others. Thank you.

**Comment by kab163** [comment]
> @kab163 please take a look at this PR I put up yesterday. I think this covers the changes in your PR plus many others. Thank you.

Well, poo, yea I think it does. I didn't realize this was already in the works! We should just merge it then - are we waiting for something?

File: toss_3_x86_64_ib/compilers.yaml
Comment (adrienbernede) [inline toss_3_x86_64_ib/compilers.yaml:42]: I see 8.0.0 below. Is that intentional?

**Review by adrienbernede — COMMENTED** [review]
Thanks for the additions @rhornung67.

Are all the compilers versions here supported by Spack@0.19.0 ? (Including cuda 12).

I'm asking because we need to know it if there is a requirement on the Spack version to use.

File: toss_3_x86_64_ib/compilers.yaml
Comment (rhornung67) [reply toss_3_x86_64_ib/compilers.yaml:42]: I checked most of the specs manually by generating a host-config file with Spack through uberenv. I may have not tested CUDA 12. I'll do that and report back. 

The clang 8.0.1 entry is a mistake. I just pushed a fix. Good catch.

Note that merging this PR is not urgent. We will be updating our CI jobs after the holidays and I wanted to get updates here done first.

Comment (rhornung67) [reply toss_3_x86_64_ib/compilers.yaml:42]: Also, I'm finishing up a RAJA Dev Guide PR that explains how all the CI stuff works and how to perform basic maintenance tasks. I will tag you as a reviewer on that and would appreciate it if you would look it over for correctness.

**Comment by rhornung67** [comment]
@davidbeckingsale when you have some time, please look this over. My hope is that after this is merged, we can meet with @adrienbernede and other relevant projects and update/minimize shared CI jobs.

**Review by kab163 — APPROVED** [review]

**Comment by kab163** [comment]
> @davidbeckingsale when you have some time, please look this over. My hope is that after this is merged, we can meet with @adrienbernede and other relevant projects and update/minimize shared CI jobs.

@rhornung67 and @adrienbernede - I'd like to merge this, is that ok?

**Comment by rhornung67** [comment]
Fine with me. @adrienbernede should approve.

File: blueos_3_ppc64le_ib/compilers.yaml
Comment (adrienbernede) [reply blueos_3_ppc64le_ib/compilers.yaml:242]: @kab163 @rhornung67 I left my review comment pending for days... :S

Comment (rhornung67) [reply blueos_3_ppc64le_ib/compilers.yaml:242]: The gcc default remains fixed on LC machines (sometimes different default for different TOSS versions) for a loooonng time. Moving to a new default is usually preceded by emails from LC indicating when the defaults will change. So we will have time to respond. 

If using GitLab to track changes to defaults will give us additional alerts, I think that is a good idea.

Comment (rhornung67) [reply blueos_3_ppc64le_ib/compilers.yaml:242]: Oh, wait. I think I mis-interpreted your question. I'll take a look and push a change if needed.

**Pull request merged**
